xla/service/gpu/fusions/transpose_mlir.cc

```diff
@@ -405,5 +410,376 @@ IndexingMap MlirTransposeFusion::GetIndexing(bool input,
   return result;
 }
 
+namespace {
+
+struct HloBytesUsageDesc {
+  int tile_size;
+  int elem_bytes;
+  int ref_count;
+  int ref_id;
+  int unused_bytes;
+};
+
+std::optional<HloBytesUsageDesc> GetHloBytesUsageDesc(
+    HloInstructionAdaptor& instr_adaptor,
+    absl::InlinedVector<int, 2>& live_range,
+    absl::InlinedVector<HloBytesUsageDesc, 2>* instr_bytes_desc,
+    absl::flat_hash_map<const HloInstruction*, int>& instr_to_ids) {
```

Comment:
Let's get rid of the `instrs_to_ids` structure and instead index the data structures directly by pointers. The added indirection doesn't seem to do anything useful here, but it's harder to read.